client/landing/jetpack-cloud/components/threat-dialog/index.tsx

```diff
@@ -2,62 +2,90 @@
 
 /**
  * Internal dependencies
  */
-import { Button, Dialog } from '@automattic/components';
+import { Button } from '@automattic/components';
 import Gridicon from 'components/gridicon';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 import ThreatItemHeader from 'landing/jetpack-cloud/components/threat-item-header';
+import ServerCredentialsWizardDialog from 'landing/jetpack-cloud/components/server-credentials-wizard-dialog';
 import { Threat } from 'landing/jetpack-cloud/components/threat-item/types';
 import { getThreatFix } from 'landing/jetpack-cloud/components/threat-item/utils';
+import getJetpackCredentials from 'state/selectors/get-jetpack-credentials';
 
 interface Props {
 	threat: Threat;
 	action: 'fix' | 'ignore';
+	siteId: number;
 	siteName: string;
 	showDialog: boolean;
 	onCloseDialog: Function;
 	onConfirmation: Function;
 }
 
-class ThreatDialog extends React.PureComponent< Props > {
-	render() {
-		const { action, onCloseDialog, onConfirmation, siteName, showDialog, threat } = this.props;
-		const isScary = action !== 'fix';
-		const buttons = [
+const ThreatDialog: React.FC< Props > = ( {
+	action,
+	onCloseDialog,
+	onConfirmation,
+	siteId,
+	siteName,
+	showDialog,
+	threat,
+} ) => {
+	const userNeedsCredentials = useSelector( ( state ) =>
+		isEmpty( getJetpackCredentials( state, siteId, 'main' ) )
+	);
+
+	const buttons = React.useMemo(
+		() => [
 			<Button className="threat-dialog__btn" onClick={ onCloseDialog }>
 				{ translate( 'Go back' ) }
 			</Button>,
-			<Button primary scary={ isScary } className="threat-dialog__btn" onClick={ onConfirmation }>
+			<Button
+				primary
+				scary={ action !== 'fix' }
+				className="threat-dialog__btn"
+				onClick={ onConfirmation }
+			>
 				{ action === 'fix' ? translate( 'Fix threat' ) : translate( 'Ignore threat' ) }
 			</Button>,
-		];
+		],
+		[ action, onCloseDialog, onConfirmation ]
+	);
 
-		return (
-			<Dialog
-				additionalClassNames="threat-dialog"
-				isVisible={ showDialog }
-				onClose={ onCloseDialog }
-				buttons={ buttons }
-			>
-				<h1
-					className={ classnames(
-						'threat-dialog__header',
-						`threat-dialog__header--${ action }-threat`
-					) }
-				>
-					{ action === 'fix'
-						? translate( 'Fix threat' )
-						: translate( 'Do you really want to ignore this threat?' ) }
-				</h1>
+	const titleProps = React.useMemo(
+		() => ( {
+			title:
+				action === 'fix'
+					? translate( 'Fix threat' )
+					: translate( 'Do you really want to ignore this threat?' ),
+			titleClassName: `threat-dialog__header--${ action }-threat`,
+		} ),
+		[ action ]
+	);
+
+	return (
+		<ServerCredentialsWizardDialog
+			siteId={ siteId }
+			showDialog={ showDialog }
+			onCloseDialog={ onCloseDialog }
+			showServerCredentialStep={ action === 'fix' && userNeedsCredentials }
+			buttons={ buttons }
+			{ ...titleProps }
+			subtitle={ translate( 'You have selected to fix a discovered threat' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'You have selected to fix all discovered threats' )` *ES Score: 6*